airflow/gcp/example_dags/example_mlengine.py

```diff
@@ -0,0 +1,219 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""
+Example Airflow DAG for Google ML Engine service.
+"""
+import os
+from typing import Dict
+
+import airflow
+from airflow import models
+from airflow.gcp.operators.mlengine import (
+    MLEngineBatchPredictionOperator, MLEngineModelOperator, MLEngineTrainingOperator, MLEngineVersionOperator,
+)
+from airflow.gcp.utils import mlengine_operator_utils
+from airflow.operators.bash_operator import BashOperator
+
+PROJECT_ID = os.environ.get("GCP_PROJECT_ID", "example-project")
+
+MODEL_NAME = os.environ.get("GCP_MLENGINE_MODEL_NAME", "model_name")
+
+SAVED_MODEL_PATH = os.environ.get("GCP_MLENGINE_SAVED_MODEL_PATH", "gs://test-airflow-mlengine/saved-model/")
+JOB_DIR = os.environ.get("GCP_MLENGINE_JOB_DIR", "gs://test-airflow-mlengine/keras-job-dir")
+PREDICTION_INPUT = os.environ.get("GCP_MLENGINE_PREDICTION_INPUT",
+                                  "gs://test-airflow-mlengine/prediction_input.json")
+PREDICTION_OUTPUT = os.environ.get("GCP_MLENGINE_PREDICTION_OUTPUT",
+                                   "gs://test-airflow-mlengine/prediction_output")
+TRAINER_URI = os.environ.get("GCP_MLENGINE_TRAINER_URI", "gs://test-airflow-mlengine/trainer.tar.gz")
+TRAINER_PY_MODULE = os.environ.get("GCP_MLENGINE_TRAINER_TRAINER_PY_MODULE", "trainer.task")
+
+SUMMARY_TMP = os.environ.get("GCP_MLENGINE_DATAFLOW_TMP", "gs://test-airflow-mlengine/tmp/")
+SUMMARY_STAGING = os.environ.get("GCP_MLENGINE_DATAFLOW_STAGING", "gs://test-airflow-mlengine/staging/")
+
+default_args = {
+    "start_date": airflow.utils.dates.days_ago(1),
+    "params": {
+        "model_name": MODEL_NAME
+    }
+}
+
+with models.DAG(
+    "example_gcp_mlengine", default_args=default_args, schedule_interval=None  # Override to match your needs
+) as dag:
+    training = MLEngineTrainingOperator(
+        task_id="training",
+        project_id=PROJECT_ID,
+        region="us-central1",
```

Comment:
It would be better to make this as an env variable, too.

Reply:
Recently, we have been trying to limit the creation of environment variables to a minimum.  We create them only to provide isolation between different tests.